tokio/src/runtime/task/mod.rs

```diff
@@ -50,20 +50,68 @@ unsafe impl<S> Sync for Task<S> {}
 #[repr(transparent)]
 pub(crate) struct Notified<S: 'static>(Task<S>);
 
+/// A task not yet bound to an executor. This object holds two ref-counts to
+/// the task to enable it to be split into two for free.
+pub(crate) struct UnboundTask<T, S: 'static> {
```

Comment:
Could you remind me why we have `UnboundTask` vs. immediately bind in spawn?

Reply:
It is the more direct change to do it like this, but I will think about whether I can remove it without changing too much code.